### ISSUE TITLE
bump to IBC v7

### ISSUE THREAD

**Issue by evan-forbes** [celestiaorg/celestia-app#2579]
Currently we use IBC v6, but other chains such as penumbra require v7 for support.

**Comment by hdevalence** [comment]
Hey! Some clarification might be helpful here.

Penumbra uses the Jellyfish Merkle Tree (similar to Celestia's SMT design) as its merkle tree, including for its IBC store.

We were able to write an ICS23-compatible proof spec for inclusion proofs without any changes to ICS23. However, for _nonexistence_ proofs, we realized that the original framework for nonexistence proofs in ICS23 only worked with IAVL-like trees that preserve key ordering.  To fix this, we upstreamed a new `prehash_key_before_comparison` field to the ICS23 proof spec standard that allows non-inclusion proofs to work with trees like the JMT and SMT. We believed that this would be a forward-compatible change, because of the protobuf behavior that ignores unknown fields in protos. Unfortunately this turns out not to be the case, because the Cosmos SDK explicitly rejects proof specs with unknown fields. More details on this can be found in our tracking issue: https://github.com/penumbra-zone/penumbra/issues/3045

So, Penumbra doesn't have any specific version requirement for `ibc-go`. However, we do have a requirement on `ics23` `0.10.0`: timeouts will only work if we construct the connection with a counterparty running `ics23` `0.10.0` or greater.

My understanding is that changing from ICS23 `0.9.0` to `0.10.0` doesn't involve any external API changes and could be accomplished without updating `ibc-go` using a patch directive.

Without wanting to make demands of the team pre-mainnet, if it was possible to get ICS23 in before mainnet, we'd be able to have fully-functional IBC between Celestia and Penumbra. Otherwise, we'd be in the difficult position of having either (1) no connection or (2) a connection that cannot handle timeouts (and hence return undelivered funds) and would require a manual upgrade / state change to enable them in the future.
